File: javascript/7kyu/Nth power rules them all.js
// https://www.codewars.com/kata/58aed2cafab8faca1d000e20

// You are provided with an array of positive integers and an additional integer n (n > 1).
// Calculate the sum of each value in the array to the nth power. Then subtract the sum of the original array.

function modifiedSum(a, n) {
    var result=0
    for(var x = 0 ; x<a.length;x++){
      result+=Math.pow(a[x],n)-a[x]
    }
  return result
}

File: java/7kyu/Nth power rules them all.java
// https://www.codewars.com/kata/58aed2cafab8faca1d000e20

// You are provided with an array of positive integers and an additional integer n (n > 1).
// Calculate the sum of each value in the array to the nth power. Then subtract the sum of the original array.
// Examples:
// [1, 2, 3] >> 6
// (1^6 - 1) + (2^6 - 2) + (3^6 - 3) = 6
// [1, 3, 4] >> 11
// (1^6 - 1) + (3^6 - 3) + (4^6 - 4) = 11

public class Kata {
  public static int modifiedSum(int[] array, int power) {
   int sum = 0 ; 
     for (int x : array) 
       sum = sum + (int)Math.pow(x,power) - x ; 
    return sum ;
    }
}